### PR TITLE
MB-15610 - Add sit playwright tests

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This PR should introduce two testharness scenarios that create moves with shipments in SIT.

Something to pay extra attention to: I moved a bunch of `waitForPage` functions from the playwright/tests/office/servicescounseling/servicesCounselingTestFixture.js to playwright/tests/utils/officeTest.js since these pages are shared between all office users, and pages that implement `OfficePage` can add their own pages for specific user roles.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. check out the new testharness scenarios by visiting the testharness url
/testharness/list
new scenarios are named `HHGMoveIn200DaysSIT` and `HHGMoveIn200DaysSITWithPendingExtension`
2. run the new SIT playwright tests from a console window
`yarn playwright test playwright/tests/office/txo/sitUpdates.js`

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: pkg/testdatagen/testharness/make_move.go [hunk 1]
```diff
@@ -3471,3 +3471,526 @@ func MakeApprovedMoveWithPPMShipmentAndExcessWeight(appCtx appcontext.AppContext
...
+			},
+		},
+		{
+			Model:    mto,
+			LinkOnly: true,
+		},
+	}, nil)
+
+	agentUserInfo := newUserInfo("agent")
+	testdatagen.MakeMTOAgent(appCtx.DB(), testdatagen.Assertions{
```
Comment (mlm55) [inline pkg/testdatagen/testharness/make_move.go:3567]: Suggestion: could we switch to using `factory.BuildMTOAgent`

Comment (michaelvaldes-truss) [reply pkg/testdatagen/testharness/make_move.go]: Fixed!

File: playwright/tests/utils/testharness.js [hunk 2]
```diff
@@ -140,6 +140,20 @@ export class TestHarness {
...
+    return this.buildDefault('HHGMoveIn200DaysSIT');
+  }
+
+  /**
+   * Use testharness to build a move with an hhg shipment in SIT and a pending SIT extension
+   */
+  async buildHHGMoveIn200DaysSITWithPendingExtension() {
+    return this.buildDefault('HHGMoveIn200DaysSITWithPendingExtension');
+  }
+
```
[the remark this answers is not in the document]
Comment (michaelvaldes-truss) [reply playwright/tests/utils/testharness.js:158]: Fixed in the testharness!

File: playwright/tests/utils/officeTest.js [hunk 3]
```diff
@@ -50,6 +42,47 @@ export class OfficePage extends BaseTestPage {
...
+    reviewProGear: async () => {
+      await base.expect(this.page.getByRole('heading', { name: 'Review pro-gear 1', level: 3 })).toBeVisible();
+    },
+    reviewReceipt: async () => {
+      await base.expect(this.page.getByRole('heading', { name: 'Review receipt 1', level: 3 })).toBeVisible();
+    },
+    reviewDocumentsConfirmation: async () => {
+      await base.expect(this.page.getByRole('heading', { name: 'Send to customer?', level: 3 })).toBeVisible();
+    },
+  };
```
[the remark this answers is not in the document]
Comment (michaelvaldes-truss) [reply playwright/tests/utils/officeTest.js]: That sounds great! I ended up extending WaitForPage into three subclasses (one for each app). Lmk if I should change this any more!

On a related note, I think I'd like to organize the test pages (e.g. officeTest.js) and the WaitForPage files (e.g. waitForOfficePage.js) in the utils directory into directories based on app (like the tests are organized; my, office, and admin), but I'll do that in a separate PR in order to keep this one more readable.

**Review by ahobson — APPROVED** [review]
This is awesome stuff! Thank you for going the extra mile!

**Review by afoon — APPROVED** [review]
LGTM

**Pull request merged**